model/gru.py

```diff
@@ -46,21 +46,36 @@ def init_hidden(self):
             earlier versions use "Variable" to initiate tensor
             variable
         '''
-        self.hidden_layer = tensor(torch.zeros(self.nb_lstm_layers,
+        self.hidden_layer = Variable(torch.zeros(self.nb_lstm_layers,
                     self.batch_size, self.nb_lstm_units))
 
-    def forward(self, batch_data):
+    def forward(self, batch_data, seq_lens):
         '''
             apply the forward function for the model
+            clarifying with example from:
+            towardsdatascience.com/\
+            taming-lstms-variable-sized-mini-batches-and-why-pytorch-is-good-for-your-health-61d35642972e
         '''
-        seq_len = batch_data.size()[1]
-        # check that the gru unit treats the s
-        gru_out, self.hidden = self.model(batch_data, self.hidden)
-        output = self.hidden_to_output(gru_out)
```

Comment:
@zhampel whenever you have a chance it would be great to get your thoughts on how I structure this GRU model. 

I'm organizing the forward prop as follows:    

   gru_out =  GRU(input)
   linear_output = Linear(gru_out)
   output = sigmoid(linear_output)